data_structures/src/Bag/Bag.cs

```diff
@@ -6,41 +6,80 @@ namespace ADT
 {
     public class Bag<T>: IBag<T>
     {
-        public Bag(int size = 10) { }
+        private T[] _bag { get; set; }
+        private int _fullSpots { get; set; }
+
+        public Bag(int size = 10) { 
+            _bag = new T[size];
+        }
 
         public void Add(T item)
         {
-            throw new NotImplementedException();
+            try
```

Comment:
No need for `try/catch`. You should throw an exception if the bag is full however.